.github/workflows/test-build-rpm.yml

```diff
@@ -156,236 +156,199 @@ jobs:
+
+  Test-SQL-NoSec:
+    needs: [Provision-Runners]
+    name: Test-SQL-NoSec
+    runs-on: [self-hosted, Linux, X64, odfe-rpm-sql-nosec]
+    steps:
+      - uses: actions/checkout@v1
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      - name: Starting ES service and retrieve tags
+        run: echo ::set-env name=p_tag_sql_nosec::$(.github/scripts/plugin_tag.sh opendistro-for-elasticsearch/sql); .github/scripts/setup_runners_service.sh --es-nosec
+      - uses: actions/checkout@v1
         with:
           repository: opendistro-for-elasticsearch/sql
-          ref: ${{env.p_tag_sql}}
-      
-      - name: IT for SQL
-        if: always()
+          ref: ${{env.p_tag_sql_nosec}}
+      - name: IT for SQL NoSec
         run: |
-          curl -XGET http://127.0.0.1:9200
-          export PATH=$PATH:$JAVA_HOME
-          cd $GITHUB_WORKSPACE/../sql
+          curl -XGET http://127.0.0.1:9200; export PATH=$PATH:$JAVA_HOME; cd $GITHUB_WORKSPACE/../sql
           ./gradlew integTest -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername=es-integrationtest
-          
-      - name: Checking out AD repo
-        if: always()
-        uses: actions/checkout@v1
+
+  Test-KNN-NoSec:
+    needs: [Provision-Runners]
+    name: Test-KNN-NoSec
+    runs-on: [self-hosted, Linux, X64, odfe-rpm-knn-nosec]
+    steps:
+      - uses: actions/checkout@v1
+      - name: Setup Java
+        uses: actions/setup-java@v1
         with:
-          repository: opendistro-for-elasticsearch/anomaly-detection
-          ref: ${{env.p_tag_ad}}
-          
-      - name: IT for AD
-        if: always()
-        run: |
-          curl -XGET http://127.0.0.1:9200
-          export PATH=$PATH:$JAVA_HOME
-          cd $GITHUB_WORKSPACE/../anomaly-detection
-          ./gradlew :integTestRunner --tests "com.amazon.opendistroforelasticsearch.ad.rest.*IT" --tests "com.amazon.opendistroforelasticsearch.ad.e2e.*IT" -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername=es-integrationtest
-      - name: Checking out k-NN repo
-        if: always()
-        uses: actions/checkout@v1
+          java-version: 14
+      - name: Starting ES service and retrieve tags
+        run: echo ::set-env name=p_tag_knn_nosec::$(.github/scripts/plugin_tag.sh opendistro-for-elasticsearch/k-NN); .github/scripts/setup_runners_service.sh --es-nosec
+      - uses: actions/checkout@v1
         with:
           repository: opendistro-for-elasticsearch/k-NN
-          ref: ${{env.p_tag_knn}}
-          
-      - name: IT for KNN
-        if: always()
+          ref: ${{env.p_tag_knn_nosec}}
+      - name: IT for KNN NoSec
         run: |
-          curl -XGET http://127.0.0.1:9200
-          export PATH=$PATH:$JAVA_HOME
-          cd $GITHUB_WORKSPACE/../k-NN
+          curl -XGET http://127.0.0.1:9200; export PATH=$PATH:$JAVA_HOME; cd $GITHUB_WORKSPACE/../k-NN
```

Comment:
Why do we make this change? I think multi lines is more readable

Reply:
The workflow is way too large......